### PR TITLE
[Docs] `jsx-no-useless-fragment`: add more examples of correct code

### PR DESCRIPTION
**Pull request by karlhorky**
Hey @ljharb ! 👋

Just a quick PR to:

1. Add some simpler examples of how to resolve the `react/jsx-no-useless-fragment` problems when they appear (there are not as many simple examples of just removing the fragment when there's a single child)
2. Add an example with an embedded expression and ternary

### CODE REVIEW

File: docs/rules/jsx-no-useless-fragment.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ Examples of **incorrect** code for this rule:
 Examples of **correct** code for this rule:
 
 ```jsx
+{foo}
+
+<Foo />
```
Comment (ljharb) [inline docs/rules/jsx-no-useless-fragment.md:37]: i'm confused, why are these needed to make the following fragment useless?

Comment (karlhorky) [reply docs/rules/jsx-no-useless-fragment.md:39]: ah sorry if this was unclear, they were intended to be the simplifications for these (the first two examples in the incorrect code):

```jsx
<>{foo}</>

<><Foo /></>
```

**Review by ljharb — APPROVED** [review]

**Comment by karlhorky** [comment]
Thanks for the review and merge!
